Sofiar/generate sdk tendlc capabilities

sofiar-msft opened the pull request:
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

sofiar-msft committed:
Generate new autorest

sofiar-msft committed:
Fix issues

sofiar-msft committed:
Add directive to rename Error to ErrorMessage

sofiar-msft committed:
Update version

sofiar-msft committed:
Update recordings

sofiar-msft committed:
Add new version to client options

sofiar-msft committed:
Remove change from generated models

sofiar-msft committed:
Remove change from generated models

sofiar-msft committed:
Add unwanted file delete

sofiar-msft committed:
Update client options to use new api verion

sofiar-msft committed:
Update assets.json

sofiar-msft committed:
Update assets.json

sofiar-msft committed:
Update assets.json

sofiar-msft committed:
Merge branch 'main' of https://github.com/Azure/azure-sdk-for-net into sofiar/generate_sdk_tendlc_capabilities

sofiar-msft committed:
Update recordings

sofiar-msft committed:
Regenerate client

sofiar-msft committed:
Merge branch 'main' of https://github.com/Azure/azure-sdk-for-net into sofiar/generate_sdk_tendlc_capabilities

sofiar-msft committed:
run ExportAPI

sofiar-msft committed:
Update PhoneNumberSearchResultErrorTests to use ErrorMessage

sofiar-msft committed:
Merge branch 'main' of https://github.com/Azure/azure-sdk-for-net into sofiar/generate_sdk_tendlc_capabilities

sofiar-msft committed:
Add autorest version

sofiar-msft committed:
Update samples

sofiar-msft committed:
Merge branch 'main' of https://github.com/Azure/azure-sdk-for-net into sofiar/generate_sdk_tendlc_capabilities

sofiar-msft committed:
Update samples

sofiar-msft committed:
Export API

sofiar-msft committed:
Merge from preview

sofiar-msft committed:
Fix build errors

sofiar-msft committed:
Run scripts

sofiar-msft committed:
Update recordings

sofiar-msft committed:
Restore recordings

sofiar-msft committed:
Merge branch 'release/azure-communication-phonenumbers/preview-main' of https://github.com/Azure/azure-sdk-for-net into sofiar/generate_sdk_tendlc_capabilities

seanmcc-msft committed:
Storage STG 93 (#41472)

sarkar-rajarshi committed:
[ACS JobRouter] Worker maxconcurrent offers (#41802)

* missing usage of Azure.Etag in RouterQueueItem

* feautre max concurrent workers

* update CreateWorkerOptions

* regen code

* add sample for max concurrent offer on worker

* update sessions records and changelog

* add missing test session recods

* rebase with main

azure-sdk commented:
**API change check**

APIView has identified API level changes in this PR and created following API reviews.

[Azure.Communication.PhoneNumbers](https://apiview.dev/Assemblies/Review/e465f73b881e41a5875a9d614a96d3ad?revisionId=02f9830f99e84dc4baad48a67e3ef0a0)

m-nash committed:
Remove references to optional in other shared source files (#42263)

* remove references to optional in other shared source files

* update communication custom code

sofiar-msft committed:
Update phone numbers clients UpdateCapabilities

glorialimicrosoft committed:
Update the changelog content and date in preparation for GA (#42266)

* Update the changelog content and date in preparation for GA

* remove  --prerelease tag from readme

* fix - The date for the changelog being released must be the latest in the file.

m-nash committed:
Remove reference to Argument in shared source (#42268)

* remove reference to Argument in shared source

* remove extra using

* add missing checks

archerzz committed:
fix(adf): `DataFactoryResource.GetPipelineRuns` cannot do pagination (#42247)

- add customization codes to implement the non-standard pagination logic of
`DataFactoryResource.GetPipelineRuns`
- add test

fix #39438

---------

Co-authored-by: Mingzhe Huang (from Dev Box) <mingzhehuang@microsoft.com>

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20240228.3 (#42277)

* Update Generator Version 3.0.0-beta.20240228.3

* Update SDK codes sq_wo_6

* Update SDK codes co_da_1

jsquire committed:
[Service Bus] Clear diagnostic properties on clone (#42261)

* [Service Bus] Clear diagnostic properties on clone

The focus of these changes is to ensure that the
diagnostic properties of `ServiceBusReceivedMessage`
are cleared when it is used to create a new
`ServiceBusMessage`.  Previously, the diagnostic
id was erroneously kept, making it look like a single
message was being received and sent multiple times.

jsquire committed:
[CODEOWNERS] Metadata fix (#42288)

Fixing a mistake that left a path segment where `ServiceOwners` should appear

annelo-msft committed:
Preparation for release of System.ClientModel v1.1.0-beta.2 (#42267)

* update CHANGELOG

* fix

drielenr committed:
[DevCenter] Add suppressions for models (#42292)

glorialimicrosoft committed:
Update the typespec to the one in Main branch (#42296)

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools for PR 7682 (#42207)

* Support resolving environment variable references in matrix config

* Improve type and null handling

* Fix reference bug

* Change behavior on missing env vars to throw

---------

Co-authored-by: Ben Broderick Phillips <bebroder@microsoft.com>

azure-sdk committed:
Increment package version after release of System.ClientModel (#42300)

azure-sdk committed:
Update github-event-processor ver to 1.0.0-dev.20240229.2 (#42306)

Co-authored-by: James Suplizio <jasupliz@microsoft.com>

eatkimchee committed:
Fist iotfirmwaredefense update SDK to swagger 2024-01-10 (#42005)

azure-sdk committed:
Update Generator Version 3.0.0-beta.20240229.1 (#42311)

pshao25 committed:
update openai (#42280)

dvbb committed:
[Dns] Add model factory for `FooRecordData` (#41680)

ArthurMa1978 committed:
Fix the spec address (#42313)

melina5656 committed:
Add operation group 'ManagementGroupSubscriptions' (#41173)

azure-sdk committed:
Increment package version after release of Microsoft.Azure.Batch (#42295)

azure-sdk committed:
Increment package version after release of Azure.Communication.Messages (#42301)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.IotFirmwareDefense (#42318)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager (#42319)

Dan-127001 committed:
Update TableSasBuilder.RowKeyEnd summary (typo) (#42321)

thang-bit committed:
Add EASM .NET data plane SDK (#40595)

* Add EASM

* Add Data Connection test

* Add DiscoveryGroupsTest.cs

* Add DiscoTemplatesTest

* Add ReportsTest.cs

* Remove white spaces in tests

* Add samples

* Update samples

* update readme

* Update AssetResource View sample

* Add more samples

* Fix spelling

* Run samples update

* Add api folder

* Update list expression in sample snippet

* Update savedfilters sample get saved filters method

* Update DiscoveryGroups sample

* Update snippet again

* Update snippets

* Record tests

* Sync with main

* Regenerate .NET sdk

* Update samples GetType() function

* Update genrated code

* Update with main

* Add PR owners to EASM

* Remove discovery group name

* Address sdk review comments

* Remove codeowners

* re-generate sdk

* Re-generate sdk

* Update with main branch

* Update, samples, readme, and model names

* sync sdk

* Merge branch 'main' of https://github.com/Azure/azure-sdk-for-net into easm

* Remove unused package-lock.json

* Update sdk/easm/Azure.Analytics.Defender.Easm/README.md

Co-authored-by: Jesse Squire <jesse.squire@gmail.com>

* Update sdk/easm/Azure.Analytics.Defender.Easm/README.md

Co-authored-by: Jesse Squire <jesse.squire@gmail.com>

* Update sdk/easm/Azure.Analytics.Defender.Easm/README.md

Co-authored-by: Jesse Squire <jesse.squire@gmail.com>

* Update documentation

* Add service owner

* Remove unused comments in AssemblyInfo.cs

* Add EASM as label

* Update readme instructions, fix code owners linting errors, and update sample titles

* PUtting under SDK region and use PRLabel

* Add CHANGELOG.md

* Run prepare release pipeline

* Switch the label to Defender EASM

* Update sdk/easm/Azure.Analytics.Defender.Easm/CHANGELOG.md

Co-authored-by: Jesse Squire <jesse.squire@gmail.com>

* Update sdk/easm/Azure.Analytics.Defender.Easm/samples/Sample1_DiscoveryGroups.md

Co-authored-by: Jesse Squire <jesse.squire@gmail.com>

* Update sdk/easm/Azure.Analytics.Defender.Easm/samples/Sample4_SavedFilters.md

Co-authored-by: Jesse Squire <jesse.squire@gmail.com>

* Update sdk/easm/Azure.Analytics.Defender.Easm/samples/Sample3_DiscoveryTemplates.md

Co-authored-by: Jesse Squire <jesse.squire@gmail.com>

* Update sdk/easm/Azure.Analytics.Defender.Easm/samples/Sample4_SavedFilters.md

Co-authored-by: Jesse Squire <jesse.squire@gmail.com>

* Update sdk/easm/Azure.Analytics.Defender.Easm/README.md

Co-authored-by: Jesse Squire <jesse.squire@gmail.com>

* Update .github/CODEOWNERS

Co-authored-by: Jesse Squire <jesse.squire@gmail.com>

* Update sdk/easm/Azure.Analytics.Defender.Easm/README.md

Co-authored-by: Jesse Squire <jesse.squire@gmail.com>

* Update sdk/easm/Azure.Analytics.Defender.Easm/README.md

Co-authored-by: Jesse Squire <jesse.squire@gmail.com>

* Sentencing casing readme title

* sentence casing

* Fix sentence casing

* add microsoft defender taxonomy

---------

Co-authored-by: Jesse Squire <jesse.squire@gmail.com>

azure-sdk committed:
Fix matrix CI tests (#41996)

Co-authored-by: Ben Broderick Phillips <bebroder@microsoft.com>

jsquire committed:
[Event Hubs] Change trigger checkpointing behavior (#42299)

* [Event Hubs] Change trigger checkpointing behavior

The focus of these changes is to change the behavior
of the Event Hubs trigger to not checkpoint when the
listener is shutting down. This is necessary to
prevent potential data loss from occurring when
shutting down Function retries.  Because the trigger
cannot know if the Function host would have retried
a failure if it were not shutting down, we cannot
assume that it is safe to checkpoint.  This change
ensures that the batch of events being processed
when shut down was requested will be retried by
another instance or the next time the Function app
is run.  Though this may cause duplicate processing,
it is consistent with the Event Hubs at-least-once
guarantee.

jsquire committed:
[CODEOWNERS] Minor fixes (#42325)

* [CODEOWNERS] Minor fixes

The focus of these changes is to fix spacing and put TimeSeriesInsights in the right place, collapsing the two versions of the tag that were present.

azure-sdk committed:
Update Generator Version 3.0.0-beta.20240302.1 (#42339)

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20240303.1 (#42354)

* Update Generator Version 3.0.0-beta.20240303.1

* Update SDK codes hy_mi_3

* Update SDK codes de_he_2

* Update SDK codes ad_co_0

* Update SDK codes sq_wo_6

* Update SDK codes mi_pu_4

* Update SDK codes co_da_1

* Update SDK codes pu_sq_5

* remove optional from shared source reference in libraries

* update custom code to use new generated source

---------

Co-authored-by: m-nash <64171366+m-nash@users.noreply.github.com>

jsquire committed:
[Event Hubs] 2024 March Release Prep (#42362)

The focus of these changes is to prepare the core
Event Hubs library for the March 2024 release.

jsquire committed:
[Service Bus] 2024 March Release Prep (#42364)

The focus of these changes is to prepare the core
Service Bus library for its March 2024 release.

azure-sdk committed:
Fix EnforceMaxLifeOfIssues event and run time (#42366)

Co-authored-by: James Suplizio <jasupliz@microsoft.com>

ninghu committed:
Azure OpenAI: 2024-02-15-preview support  (#41859)

* OpenAI: Update baseline SDK for supporting 2024-02-15-preview

* Add customization code for TTS

* Update baseline SDK and handle chat extension changes

* Seperate s_deploymentEntriesByScenario to enchance the test execution

* Update TTS docs

* audio transcription/translation: default to .wav, provide xml doc remarks

* Regenerate baseline SDK and address completion test failures

* test recording update (including better recording mode variable sets)

* Tweak doc for AudioSpeechOptions

* Add initial content to README and CHANGELOG

* test: Add a test for GetAudioSpeech

* doc fix

* test: Exercise optional parameters in test

* test: Assert that response?.Value is not null

* changelog update

* docs: Add a sample for speech generation

* docs: Add speech generation snippet to readme

* test: Rename Sample Method Name

* test: Move variable declaration to make sample more concise

* test: Remove blank lines

* test: Remove variable declaration

* test: Remove more blank lines

* docs: Update snippets

* docs: Update snippets

* Suppress unwanted apis

* Update test recordings to include TTS test

* TTS renaming and test recording update

* Fix the CI failures

* Update generated code

* Link fix

* update baseline sdk

* Update test and CHANGELOG for the new TSP changes.

* update

* feat: Enable serialization of logprobs and top_logprobs

* tests: Add logprob completion test

* test: Add a test for top_logprobs

* chore: Update assets.json

* test: Update CanGenerateImages test to validate the presence of filter results

* tests: Use == instead of is

* tests: Add missing 'Not'

* test: More exhaustively validate filter results

* test: Remove WriteNull from custom serialization

* tests,refactor: Rename i -> result

* tests,refactor: Use variable to make assertion more clear

* tests: Remove quasi-duplicate test

* chore: Update assets.json

* update test recordings

* Fix the test

* update

---------

Co-authored-by: Travis Wilson <35748617+trrwilson@users.noreply.github.com>
Co-authored-by: kdestin <101366538+kdestin@users.noreply.github.com>

trrwilson committed:
final changelog snap for beta.14 (#42372)

m-nash committed:
put back explicit usings (#42373)

ShivangiReja committed:
[Search] Add support for `sbyte` and `int16` types to FieldBuilder (#42378)

* [Search] Add support for `sbyte` and `int16` types to FieldBuilder

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools for PR 7758 (#42381)

* Add 1es changes for job/matrix generation and publish

* Use more flexible pool filter for prev/next pool matches

* consolidate displayname definition

* use linux pool variables for generate matrix job

* Fix publish artifact

* Use single publish task for publish artifact

---------

Co-authored-by: Ben Broderick Phillips <bebroder@microsoft.com>

Commit:
[LiveMetrics] report process metrics `CPU Total` and `Committed Memory` (#42213)

* implement process counters

* use Demo for testing. REVERT THIS BEFORE MERGING

* fix typo

* changelog

* Revert "use Demo for testing. REVERT THIS BEFORE MERGING"

This reverts commit 38bf3f3a83eee996dc489643121bd2a472035cfb.

* cleanup

* update changelog

* refactor

* typo

* rename

* changelog

* rename

* pr feedback

azure-sdk committed:
Failed Update AutoRest C# version to 3.0.0-beta.20240304.1 (#42382)

* Update Generator Version 3.0.0-beta.20240304.1

* Update SDK codes de_he_2

* Update SDK codes hy_mi_3

* Update SDK codes ad_co_0

* Update SDK codes sq_wo_6

* Update SDK codes mi_pu_4

* Update SDK codes pu_sq_5

* remove changetrackinglist

* fix batch projects

* update personalizer

* regen projects from the failed ci leg

---------

Co-authored-by: m-nash <64171366+m-nash@users.noreply.github.com>

devinlstoen committed:
dotnet sdk update support api 2023 (#42308)

* dotnet sdk update support api 2023

* LookUpResourceIdResponse method names to LookUpResourceIdResult

* dotnet sdk update support api 2023

* dotnet sdk update support

* dotnet sdk update support api

* Some tweaks

* Some tweaks

* update changelong

---------

Co-authored-by: Arthur Ma <mars.arthur@263.net>

wiboris committed:
Updated Batch Track 2 Management plane SDK for 2024-02-01 API version (#41874)

* Adding support for 2024-02-01 api

* updated generation

archerzz committed:
fix(batch): orphan commit id in spec url (#42396)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.Batch (#42399)

azure-sdk committed:
Increment package version after release of Azure.AI.OpenAI (#42376)

azure-sdk committed:
Increment package version after release of Azure.Messaging.EventHubs (#42408)

annelo-msft committed:
ClientModel: Updates to ClientModel public APIs from Azure.Core 2.0 integration (#42328)

* Updates to ClientModel public APIs from Azure.Core 2.0 integration

* Enable setting message property to null

MikeyMCZ committed:
[Text Translation] Update clientRequestId parameter type to System.Guid (#42238)

* [Text Translation] Update type of clientTraceId to Guid

* Suppress the clientTraceId of type string methods

* Fix build

---------

Co-authored-by: Michal Materna <mimat@microsoft.com>

azure-sdk committed:
Increment package version after release of Azure.Analytics.Defender.Easm (#42367)

azure-sdk committed:
Increment package version after release of Azure.Messaging.ServiceBus (#42413)

jsquire committed:
[Messaging] Companion package 2024 March Release Prep (#42365)

The focus of these changes is to prepare the companion
packages for Event Hubs and Service Bus for their
March 2024 release.

azure-sdk committed:
Increment package version after release of Microsoft.Azure.WebJobs.Extensions.ServiceBus (#42415)

azure-sdk committed:
Increment package version after release of Microsoft.Azure.WebJobs.Extensions.EventHubs (#42417)

azure-sdk committed:
Increment package version after release of Azure.Messaging.EventHubs.Processor (#42418)

hallipr committed:
Initialize autorest before calling GenerateCode (#42389)

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools for PR 7810 (#42424)

* Run publish on failed or succeeded

* Expand agent os string detection

* Check agent job status env var for artifact name detection

* Add sbomEnabled flag to publish template

* Fix image and artifact name conditional

---------

Co-authored-by: Ben Broderick Phillips <bebroder@microsoft.com>

kinelski committed:
[DocumentIntelligence] Merging 1.0.0-beta.2 features (#42420)

* 2024-02-29-preview: generated from the latest spec (#2056)

* Incorporating architect feedback (#42317)

* Regenerated to incorporate latest generation changes

JoshLove-msft committed:
Prepare for GA (#42371)

* Prepare for GA

* other changes

kinelski committed:
[DocumentIntelligence] 1.0.0-beta.2: Preparing for release (#42428)

azure-sdk committed:
Increment package version after release of Azure.Search.Documents (#42431)

ArthurMa1978 committed:
Update the support RP according to the latest CodeGen change (#42430)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.Support (#42439)

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20240305.2 (#42451)

koen-mertens committed:
SDK implementation for Radiology Insights (in public preview) (#41974)

* init

* Radiology insights initial

* hero scenario test case

* Hero scenario test case

* Ranamed hero scenario to RadiologyInsightsClientSample and move it to the samples directory

* fix: token-expected: ';'

* Revert "Merge branch 'asaflevi/healthinsights-radiologyinsights-net-sdk' of https://github.com/asaflevi-ms/azure-sdk-for-net into asaflevi/healthinsights-radiologyinsights-net-sdk"

This reverts commit f19019351e575e142440e699ffc98a9bbbc8cbc5, reversing
changes made to f93993ee6962722c00de52cedcd0ee2ad34be559.

* changes for CriticalResult sdk

* Compliance with requirements for RadiologyInsights .NET SDK

* changed return type for async call

* changes for sdk live tests

* more inference samples

* corrected offsets

* changes for inferences

* followupcommunication inference

* Update README.md

* Update README.md

remove links that don't exist yet

* recorded tests

* remove samples

* clean up samples

rework samples and readme files

* add readme samples and adjust tsp-location

- change the commit id to be in sync with the latest version.
- add readme files for the 2 samples

* Update README.md

* generated files using TypeSpec-Project-Generate.ps1

* changes after running sync, generate and export scripts

* changes for providing endpoint and key from environment

* adding assets for net

* Update CHANGELOG.md

* Update CODEOWNERS

* Update CHANGELOG.md

* After sync with latest and regenerate the SDK

* removed unnecessary generated and suppressed

---------

Co-authored-by: Asaf Levi <asaflevi@microsoft.com>
Co-authored-by: rameezshaikh <rameezshaikhmicrosoft@gmail.com>

koen-mertens committed:
Update README.md for release (#42456)

* Update README.md

* Update README.md

azure-sdk committed:
Increment package version after release of Azure.Health.Insights.RadiologyInsights (#42458)

azure-sdk committed:
Increment package version after release of Azure.AI.DocumentIntelligence (#42433)

christothes committed:
Mark SNI and username/password tests as manual or playbackOnly (#42427)

JoshLove-msft committed:
API updates (#42468)

* API updates

* snippets

amnguye committed:
Adding test coverage for uneven pages sizes in a page blob (#42469)

Commit:
misc cleanup (#42432)

Commit:
Start Azure SDK log forwarder in AzMon OTel distro (#42374)

* Start Azure SDK log forwarder in AzMon OTel distro

Commit:
Follow-up: upodate changelog (#42475)

kinelski committed:
[AppConfiguration] Implemented ETag-by-page feature (#42473)

Commit:
Follow-up on #42374: update AzMonDistro readme what's included - logs (#42476)

* Follow-up: update AzMonDistro readme what's included - logs

* Update sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md

Co-authored-by: Timothy Mothra <tilee@microsoft.com>

---------

Co-authored-by: Timothy Mothra <tilee@microsoft.com>

trrwilson committed:
OpenAI Assistants: merge image file ID fix from TypeSpec (#42472)

* code interpreter image fix + release prep

* export-api update

azure-sdk committed:
TestResources - Bubble up errors in Retry with pipe statements (#42480)

Co-authored-by: Ben Broderick Phillips <bebroder@microsoft.com>

DeepikaNMS committed:
Added New features - user access mgmt, service connectors to confluent .net sdk (#42077)

Commit:
[AzureMonitorDistro] update vendored code (#42479)

* update SqlClient

* update ResourceDetector

* changelog

* delete unused Vendored\Shared code

* update copyright

* update shared code

* changelog

* update changelog

seanmcc-msft committed:
Updated Storage to point at latest stable swagger filese in main (#42497)

live1206 committed:
fix recording (#42499)

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20240306.2 (#42498)

Co-authored-by: m-nash <64171366+m-nash@users.noreply.github.com>

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20240306.3 (#42500)

* Update Generator Version 3.0.0-beta.20240306.3

* Update SDK codes de_he_2

* Update SDK codes he_mi_3

* Update SDK codes ad_co_0

* Update SDK codes sq_wo_6

* Update SDK codes mi_pu_4

* Update SDK codes co_da_1

* Update SDK codes pu_sq_5

---------

Co-authored-by: Mingzhe Huang (from Dev Box) <mingzhehuang@microsoft.com>

HarveyLink committed:
[Monitor] Fix issue #41377 (#42394)

* Fix for issue 41377

* Update CHANGELOG.md

* Update MonitorConditionOperator.Serialization.cs

ArthurMa1978 committed:
Update tests for Confluent (#42504)

azure-sdk committed:
Increment package version after release of Azure.Data.AppConfiguration (#42514)

azure-sdk committed:
Increment package version after release of Azure.AI.OpenAI.Assistants (#42485)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.Confluent (#42517)

JoshLove-msft committed:
Specify 1.36 as core dependency version to unblock autorest.csharp update (#42494)

jaschrep-msft committed:
Sas preservation fix (#42471)

* preserve signed protocol order

* test

* changelog

cplacemsft committed:
Update Azure.Data.Tables migration documentation to clarify the lack of support for session consistency. (#42491)

Commit:
Azure.Monitor: fix for Microsoft.Bcl.AsyncInterfaces (#42426)

* add net6 to AspNetCore and LiveMetrics

* fix compiler issues

* test removing bcl fix from test/demo projects

* fix compiler issues in LiveMetrics

* changelog

* re-enable nullable in Filter classes

* update error message

* revert

JoshLove-msft committed:
fix naming of internal types (#42528)

azure-sdk committed:
Update Generator Version 3.0.0-beta.20240307.1 (#42539)

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20240308.1 (#42545)

* Update Generator Version 3.0.0-beta.20240308.1

* Update SDK codes co_da_1

jsquire committed:
[CODEOWNERS] Organize management path (#42547)

The focus of these changes is to move a management path registration for pull requests to the management area for organization purposes.

azure-sdk committed:
Add dotnet-api product slug to sample metadata script (#42549)

Co-authored-by: Anne Thompson <annelo@microsoft.com>

azure-sdk committed:
Sync .github/workflows directory with azure-sdk-tools for PR 7845 (#42552)

* Create a separate job for events requiring Az CLI

* Update .github/workflows/event-processor.yml

Co-authored-by: Wes Haggard <weshaggard@users.noreply.github.com>

---------

Co-authored-by: James Suplizio <jasupliz@microsoft.com>
Co-authored-by: Wes Haggard <weshaggard@users.noreply.github.com>

azure-sdk committed:
Update Prepare-Release.ps1 to handle only one previous release (#42551)

In the case there is exactly one previous release PS will return the single object and thus the Count property will not exist. Instead this change ensures that we always have a list.

Co-authored-by: Wes Haggard <weshaggard@users.noreply.github.com>

Commit:
[AzureMonitorDistro] resync vendored code (#42522)

* cleanup copyright

* update SqlClient

annelo-msft committed:
ClientModel: Add samples for System.ClientModel library (#42369)

* Samples - WIP

* README updates

* Configuration samples

* updates before generating snippets

* update snippets

* readme updates

* intermediate backup

* updates

* fix

* updates

* nit

* nit

* fix links

* updates from PR feedback

* revert engsys file

* update product

* add sample client implementation

* add input model to sample client method

* change API key in samples

* add inline comments to sample client and change defaults on HttpClient sample

* update impressions link

* restructure to address PR feedback

* nits

* nits

* nits

* small updates from PR feedback

* add comment

* rework convenience methods section in README

* more updates; add dotnet-api slug

* Add sample showing response classifier

* updates:

* reference error response configuration sample from README

* update samples README

* update md files

* show creation of BinaryContent from model in RequestOptions sample

* add examples of different way to create BinaryContent

* show protocol method implementation and message.Apply(options)

* updates

* nits

* nits

Commit:
prep release (#42557)

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20240308.2 (#42565)

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20240308.3 (#42570)

azure-sdk committed:
Increment package version after release of Azure.Core.Expressions.DataFactory (#42554)

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20240309.1 (#42578)

* Update Generator Version 3.0.0-beta.20240309.1

* Update SDK codes de_he_2

* Update SDK codes he_mi_3

* Update SDK codes ad_co_0

* Update SDK codes sq_wo_6

* Update SDK codes mi_pu_4

* Update SDK codes co_da_1

* Update SDK codes pu_sq_5

* update code

---------

Co-authored-by: ArcturusZhang <dapzhang@microsoft.com>

JoshLove-msft committed:
Prepare for release (#42581)

azure-sdk committed:
Update github-event-processor version (#42593)

Co-authored-by: James Suplizio <jasupliz@microsoft.com>

azure-sdk committed:
Update GitHubEventProcessor to 20240311.2 (#42601)

Co-authored-by: James Suplizio <jasupliz@microsoft.com>

jsquire committed:
[CODEOWNERS] Formatting fix (#42598)

The focus of these changes is to fix indentation so that formatting is consistent.

azure-sdk committed:
Increment package version after release of Azure.Monitor.OpenTelemetry.LiveMetrics (#42605)

azure-sdk committed:
Skip env processing on matrix import step (#42607)

Co-authored-by: Ben Broderick Phillips <bebroder@microsoft.com>

hallipr committed:
Use package.json for autorest in sdk repo (#42524)

nisha-bhatia committed:
[Azure.Monitor.Query] Update Swaggers and regenerate code with renames (#42414)

yifan-zhou922 committed:
azure.analytics.purview.datamap (#41739)

* CodeGen from PR 26702 in Azure/azure-rest-api-specs
Merge ec02e53e56f6daf96c70e42e89b58c1fb47a62ab into 68030f1c17e0cd5912e1bda17a5f28473e84b5f9

* update

* update

* update

* update

* update

* Update DataMapClientTest.cs

* update

* update tests

* update

* Update README.md

* update test-resource

* update tests

* update

* update

* Create assets.json

* Update assets.json

* Update ignore-links.txt

* Update README.md

* regen the models

* update

* regen the sdks

* update changelog

* change the mode to default

* Update README.md

* Update DataMapClientTestBase.cs

* update

* Update README.md

* update tests (all passed locally)

* update

* Update README.md

* update samples

* update link

* update examples

* update

* update

* update examples

* Create Azure.Analytics.Purview.DataMap.netstandard2.0.cs

* update examples

* Update sdk/purview/Azure.Analytics.Purview.DataMap/samples/Sample1_GetTypeByNameAsync.md

* Update sdk/purview/Azure.Analytics.Purview.DataMap/samples/Sample1_GetTypeByName.md

* Create Utf8JsonRequestContent.cs

* Update CHANGELOG.md

---------

Co-authored-by: SDKAuto <sdkautomation@microsoft.com>
Co-authored-by: catalinaperalta <catalinaperaltah@hotmail.com>

azure-sdk committed:
Increment package version after release of Azure.Messaging.EventGrid (#42608)

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20240311.2 (#42616)

* Update Generator Version 3.0.0-beta.20240311.2

* Update SDK codes de_he_2

* Update SDK codes he_mi_3

* Update SDK codes ad_co_0

* Update SDK codes sq_wo_6

* Update SDK codes mi_pu_4

* Update SDK codes co_da_1

* Update SDK codes pu_sq_5

* remove changetrackingdictionary

* update core

* update duplicate shared source projects

* add to fully custom code

---------

Co-authored-by: m-nash <64171366+m-nash@users.noreply.github.com>

azure-sdk committed:
Increment package version after release of Azure.Analytics.Purview.DataMap (#42619)

HarmanDhunna committed:
Support overrides of token issuer from environment variables  (#41945)

rajkumar-rangaraj committed:
[AzureMonitorDistro] Add ASP.NET Core and HTTP Client metrics instrumentation (#42307)

* Add ASP.NET Core and HTTP Client metrics instrumentation

* Changelog update

* Update .NET 8.0 and above to use built-in meter.

* Update readme

* readme - swich doc to use new api.

* Update sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md

* Update sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md

* Update CHANGELOG.md

* Update CHANGELOG.md

* Update CHANGELOG.md

* Update CHANGELOG.md

* Update sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md

Co-authored-by: Rajkumar Rangaraj <rajrang@microsoft.com>

* Update README.md

* Update README.md

* Update sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md

Co-authored-by: Vishwesh Bankwar <vishweshbankwar@users.noreply.github.com>

* Update README.md

* Update README.md

* Update README.md

* Update OpenTelemetryBuilderExtensions.cs

* Update sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md

Co-authored-by: Vishwesh Bankwar <vishweshbankwar@users.noreply.github.com>

* Update README.md

* Update README.md

* Update sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md

Co-authored-by: Vishwesh Bankwar <vishweshbankwar@users.noreply.github.com>

* Update sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/CHANGELOG.md

Co-authored-by: Vishwesh Bankwar <vishweshbankwar@users.noreply.github.com>

* Update sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md

Co-authored-by: Vishwesh Bankwar <vishweshbankwar@users.noreply.github.com>

* Update sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/README.md

Co-authored-by: Vishwesh Bankwar <vishweshbankwar@users.noreply.github.com>

---------

Co-authored-by: Timothy Mothra <tilee@microsoft.com>
Co-authored-by: Vishwesh Bankwar <vishweshbankwar@users.noreply.github.com>

christothes committed:
Fixes to PoP for InteractiveBrowserCredential Authenticate flow (#42600)

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20240311.3 (#42617)

* Update Generator Version 3.0.0-beta.20240311.3

* Update SDK codes de_he_2

* Update SDK codes he_mi_3

* Update SDK codes ad_co_0

* Update SDK codes sq_wo_6

* Update SDK codes mi_pu_4

* Update SDK codes co_da_1

* Update SDK codes pu_sq_5

* Fix tests based on new names

* fix test

* regen

* snippets

---------

Co-authored-by: Mingzhe Huang <archerzz@users.noreply.github.com>
Co-authored-by: jolov <jolov@microsoft.com>

fangchen0601 committed:
[ACS][CallAutomation]Move SourceCallerIdNumber from Answer to Transfer api (#42642)

* remove SourceCallerIdNumber from answercallOptions, update autorest with new swagger

* create internal classes for media feature and ops feature models

* update api

* add SourceCallIdNumber to Transfer

Commit:
[AzureMonitorDistro] prep AspNetCore for release (#42644)

* prep AspNetCore for release

* simplify metrics statement

nisha-bhatia committed:
wip (#42647)

azure-sdk committed:
Increment package version after release of Azure.Monitor.OpenTelemetry.AspNetCore (#42658)

azure-sdk committed:
Update Generator Version 3.0.0-beta.20240312.2 (#42661)

cochi2 committed:
Adding Hold/Unhold operations. (#42162)

* Latest swagger

* Fixing swagger

* Adding Hold/Unhold methods

* adding hold/unhold

* Maintaining StartHoldMusic backwards compatible

* Adding changes to Participants Updated event and HoldFailed

* Adding API changes

* Fixing merge with new regeneration

* Fixing test

* Fixing test

* Fixing generate errors

* Fixing merge

azure-sdk committed:
Update Generator Version 3.0.0-beta.20240313.1 (#42667)

sorrego-msft committed:
[Communication][PhoneNumbers] Fix areacodes pageable bug (#42629)

* Fix areacodes pageable bug

* Update recordings

azure-sdk committed:
APIView - send package version param for automatic API review request (#42671)

Co-authored-by: Praveen Kuttappan <prmarott@microsoft.com>

live1206 committed:
Fix for RehydrationToken serialization (#42662)

melina5656 committed:
[Authorization] Fix for issue #40050 (#42664)

wangbwn committed:
GA sdk release. (#42278)

allegradomel committed:
Aldomel/ga sdk (#42298)

azure-sdk committed:
Update Generator Version 3.0.0-beta.20240314.1 (#42688)

sumitkumams committed:
Release request for mysql api version "2023-12-01-preview" (#42624)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.DataFactory (#42689)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.Sphere (#42690)

JoshLove-msft committed:
Prepare for release (#42682)

* Prepare for release

* fix changelog

AlexanderSher committed:
Transform Rooms swagger to match customized ParticipantProperties.Role property (#42692)

* Transform Rooms swagger to match customized ParticipantProperties.Role property

* Fix comment

azure-sdk committed:
Increment package version after release of Microsoft.Azure.WebJobs.Extensions.ServiceBus (#42694)

ChristopherManthei committed:
Remote Rendering Test Fix (#42590)

* Remote Rendering Test Fix

* Error message was changed to a more user friendly one in the service.

* Updated recordings

---------

Co-authored-by: Christopher Manthei <chmant@microsoft.com>

jsquire committed:
[TestFramework] Make ClientSecret optional (#42677)

* [TestFramework] Make ClientSecret optional

The focus of these changes is to make the ClientSecret
and ClientId environment variables optional.  This is 
to support scenarios where using `DefaultAzureCredential` 
is preferred, such as local runs for live tests.

Commit:
Update OpenTelemetryBuilderExtensions.cs (#42680)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.MySql (#42695)

AlexanderSher committed:
Add nullable annotation to FormFieldsReport.accuracy (#42693)

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools for PR 7877 (#42700)

* create necessary eng/common changes to allow proxy to be shut down.

---------

Co-authored-by: Scott Beddall (from Dev Box) <scbedd@microsoft.com>

nisha-bhatia committed:
wip (#42707)

ericasp16 committed:
Convert number lookup ga to preview (#42303)

* Number format (#40083)

Add number format features to the number lookup public preview.

---------

* update autorest.md to point to the number lookup feature's latest swagger, update the auto-generated files, and add methods to the public SDK for searching operator information

* update generated files based on latest swagger

* update changelog for number lookup public preview

* update version in PhoneNumbers.csproj

* update scripts

* fix SMS changelog format

* Increment package version after release of Azure.Communication.PhoneNumbers (#38417)

* add preview version number

* update changelog

* update changelog version

* run api update script

* Increment package version after release of Azure.Communication.PhoneNumbers (#38470)

* update generated files

* implement GA number lookup updates

* updated based on SDK feedback

---------

Co-authored-by: Azure SDK Bot <53356347+azure-sdk@users.noreply.github.com>

* update generated files

* update to preview

* fix serialization of phone numbers in tests

* undo unrelated changes to SMS Changelog

* update generated files

* update generated files

* update test files, disable tests that are hanging

* update tests, run Export-API

* update tests and recordings to include all Number lookup tests

---------

Co-authored-by: Azure SDK Bot <53356347+azure-sdk@users.noreply.github.com>

dargilco committed:
Fix typos (#42712)

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20240314.3 (#42717)

* Update Generator Version 3.0.0-beta.20240314.3

* Update SDK codes de_he_2

* Update SDK codes he_mi_3

* Update SDK codes sq_wo_6

* Update SDK codes mi_pu_4

* Update SDK codes co_da_1

* Update SDK codes pu_sq_5

JoshLove-msft committed:
Always log test proxy errors (#42715)

* Always log test proxy errors

* fix names

azure-sdk committed:
Update Generator Version 3.0.0-beta.20240314.4 (#42723)

azure-sdk committed:
Increment version for communication releases (#42732)

* fix beta package version to avoid conflict

* fix beta package version to avoid conflict

* Increment package version after release of Azure.Communication.PhoneNumbers

---------

Co-authored-by: Erica Sponsler <ericasp@microsoft.com>

mcgallan committed:
Bot Service Test (#40582)

mcgallan committed:
FIx (#42056)

audunn committed:
[NetAppFiles] Bump ANF to api-version 2023-07-01 (#42546)

scottaddie committed:
Add clarity to Monitor Query client XML docs (#42742)

* Add clarity to Monitor Query client XML docs

* Fix build error

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools for PR 7892 (#42754)

* Move git check for changes to script

* Minor fixes for 1es migration

---------

Co-authored-by: Ben Broderick Phillips <bebroder@microsoft.com>

04diiguyi committed:
Azure Quantum .net SDK Update to support api key (#42291)

annelo-msft committed:
ClientModel: Add Stream creation API to BinaryContent (#42696)

* Add Stream creation API to BinaryContent

* nits

* pr fb

* tests in-progress

* updates

* call flush from WriteToAsync

* add to sync also

* nit: parameter name

scbedd committed:
Use updated proxy w/ Retry fixes for MacOS (#42758)

* point at new proxy in eng.data.props

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.NetApp (#42747)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.Quantum (#42765)

azure-sdk committed:
Update Generator Version 3.0.0-beta.20240317.2 (#42767)

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20240318.1 (#42771)

azure-sdk committed:
Create api review script changes to add a param to set release tag (#42763)

Co-authored-by: Praveen Kuttappan <prmarott@microsoft.com>

Commit:
[LiveMetrics] fix for memory not updating (#42760)

* fix for memory not updating

* changelog

nisha-bhatia committed:
[Azure.Monitor.Query] Add MetricsClientBuilderExtensions.cs (#42753)

azure-sdk committed:
Update Generator Version 3.0.0-beta.20240318.2 (#42779)

azure-sdk committed:
remove all references to docker (#42776)

Co-authored-by: Scott Beddall (from Dev Box) <scbedd@microsoft.com>

Commit:
[LiveMetrics] update demo (#42781)

* update demo

* update cpu stress test

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20240319.1 (#42796)

* Update Generator Version 3.0.0-beta.20240319.1

* Update SDK codes de_he_2

* Update SDK codes ad_co_0

* Update SDK codes he_mi_3

* Update SDK codes sq_wo_6

* Update SDK codes mi_pu_4

* Update SDK codes co_da_1

* Update SDK codes pu_sq_5

* update JsonElementExtension references

* remove the other replaced shared source

* remove the useless shared code

* add shared code for xml back in storage

* update personalizer

* fix maps

* update resourcemanager

* fix tables

---------

Co-authored-by: m-nash <64171366+m-nash@users.noreply.github.com>
Co-authored-by: ArcturusZhang <dapzhang@microsoft.com>

JoshLove-msft committed:
Add BicepModelReaderWriterOptions (#41895)

* Add BicepModelReaderWriterOptions

* Fix ref docs

* add experimental

* Fix docs

kinelski committed:
[AppConfiguration] Test fix: GetBatchSettingPagination (#42795)

scottaddie committed:
Document MetricsClient in Monitor Query README (#42799)

trrwilson committed:
address pattern-match and incompleteness bugs in function tool comparison (#42786)

nisha-bhatia committed:
[Azure.Monitor.Query] Update TROUBLESHOOTING.md (#42676)

scottaddie committed:
Update sovereign cloud heading in Monitor Query README (#42808)

azure-sdk committed:
Bump the proxy version to the last one resolving mac timeout issues (#42761)

Co-authored-by: Scott Beddall (from Dev Box) <scbedd@microsoft.com>

benbp committed:
Convert pipelines to extend from 1es pipeline templates (#42496)

* Convert pipelines to extend 1es pipeline templates

* Convert matrix files to use new pool variables

* Revert changes to old ci.mgmt.yml

* Revert matrix testing file changes

* Smoke test fixes

* Update credscan suppressions for SDLSources

* Download credscan suppressions in devops publish

* Fix smoke test dependson

* Fix credscan suppression pathing in all cases

* Fix smoke test condition

* Fix credscan pathing in some release jobs and download baselines

* Hardcode more image values in release deploy jobs

* Fix release

* Use release tag for 1es pipelines

* Try reducing size

* Updates to release to reduce size

* Revert core artifact changes

JoshLove-msft committed:
Don't bootstrap if no resource group env var (#42819)

trrwilson committed:
AOAI 2024-03-01-preview update (#42816)

HarveyLink committed:
[ResourceMover] Add operation: MoveCollections_Delete back (#42792)

* Fix issue 30952

* Update autorest.md

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20240319.5 (#42827)

* Update Generator Version 3.0.0-beta.20240319.5

* Update SDK codes de_he_2

* Update SDK codes he_mi_3

* Update SDK codes sq_wo_6

* Update SDK codes pu_sq_5

* Update SDK codes co_da_1

* Fix build

* Fix build

* Build fix

* Fix suppression

---------

Co-authored-by: Alexander Sher <alsher@microsoft.com>

xiang17 committed:
[LiveMetricsExporter] Add Filtering Support part 4 - Apply the Filters and Aggregates the result (#41843)

* Apply filter, project the metrics and count it in

* Add the filter, projection and accumulation result to the data structure to be sent

* Prepare the accumulators before applying filters and accumulating values

trrwilson committed:
Azure OpenAI: beta.15 release changelog (#42824)

* AOAI 2024-03-01-preview update

* changelog snap for beta.15

christothes committed:
Handle scenario where IMDS returns 200 but is canceled (#42809)

azure-sdk committed:
Update azure-sdk-build-tools Repository Resource Refs in Yaml files (#42844)

jsquire committed:
[CODEOWNERS] Fix baseline (#42851)

* [CODEOWNERS] Fix baseline

The focus of these changes is to fix the set of baseline errors after write permissions were updated to be added by default.

* Update CODEOWNERS_baseline_errors.txt

seanmcc-msft committed:
Update ShareClient.GetPermission/GetPermissionAsync() XML documentation (#42711)

MikeyMCZ committed:
[Text Translation] Add support for AAD for MT endpoint (#42840)

Adding a new authentication policy which is used when user is using AAD authentication and general MT endpoint. This policy is adding two required headers:
* `Ocp-Apim-ResourceId`
* `Ocp-Apim-Subscription-Region`

melina5656 committed:
[ApiManagement] Fix `BackendResponseCode` serialize issue (#39442)

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20240320.1 (#42838)

* Update Generator Version 3.0.0-beta.20240320.1

* Update SDK codes de_he_2

* Update SDK codes he_mi_3

* Update SDK codes ad_co_0

* Update SDK codes sq_wo_6

* Update SDK codes mi_pu_4

* Update SDK codes co_da_1

* Update SDK codes pu_sq_5

azure-sdk committed:
Update Generator Version 3.0.0-beta.20240320.2 (#42861)

azure-sdk committed:
Increment package version after release of Azure.AI.OpenAI (#42849)

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20240321.1 (#42872)

* Update Generator Version 3.0.0-beta.20240321.1

* Update SDK codes de_he_2

* Update SDK codes he_mi_3

* Update SDK codes mi_pu_4

* Update SDK codes co_da_1

Commit:
[LiveMetrics] small refactor (#42877)

* small refactor

* cleanup

sofiar-msft committed:
Merge from main
